### ISSUE TITLE
Exclude\Include Pointcut Attribute on multiple MethodAdvice

### ISSUE THREAD

**Issue by Kahbazi** [ArxOne/MrAdvice#89]
I have this code

```
[AdviceA, AdviceB]
    class Program
    {
        static void Main()
        {
            Console.Write("Main");
        }
    }

    [ExcludePointcut("*Main")]
    class AdviceA : Attribute , IMethodAdvice
    {
        public void Advise(MethodAdviceContext context)
        {
            Console.WriteLine("Entry From Advice A");
            context.Proceed();
            Console.WriteLine("Exit From Advice A");
        }
    }

    [IncludePointcut("*Main")]
    class AdviceB : Attribute, IMethodAdvice
    {
        public void Advise(MethodAdviceContext context)
        {
            Console.WriteLine("Entry From Advice B");
            context.Proceed();
            Console.WriteLine("Exit From Advice B");
        }
    }
```

I expect the output be like this 

```
Entry From Advice B
Main
Exit From Advice B
```

but the result is

```
Main
```

It seems the problem is when there is more than one `MethodAdviceAttribute` on class, method or assembly, Weaver would aggregate all `IncludePointcut` and `ExcludePointcut` attributes and apply them for advising.

Am I doing something wrong?

**Comment by picrap** [comment]
Hi, thanks for the clear example, I'll take a look at this shortly.

**Comment by picrap** [comment]
Fixed in 2.4.4
